### DISCUSSION TITLE
ViaPoint becomes ShapingPoint after recalculation

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#29] [Q&A] [answered]
If a recalculation is necessary (road block, route missed) the next waypoint if a via point will be changed into a shaping point. see: [https://www.youtube.com/watch?v=6O1N05yjvBA](https://www.youtube.com/watch?v=6O1N05yjvBA) 
It seems to be the same logic of changing the planned start into a shaping point if you select "start" in following selection:
<img src="https://user-images.githubusercontent.com/121806238/216101701-03745f3c-9ad0-4f95-9b4b-c1e58146220f.png" width="200"/>

**Comment by devemux86** [comment]
Thanks for the report and the video!

What is the origin of the displayed route?
What routing engine is selected in settings?

This seems to be a partial rerouting:
- Recalculate route from user location to next waypoint
- Merge the calculated part + the rest **unchanged** route

I will certainly check it.

**Comment by WolfgangZ42** [comment]
The origin was a .kurviger file, the engine is BRouter (offline) with motorcycle, shortest. 
Recalculation to next waypoint is fine and the remaining route is unchanged - but the waypoint should remain a via point.

**Comment by devemux86** [comment]
Thanks for the report!

> Recalculation to next waypoint is fine and the remaining route is unchanged - but the waypoint should remain a via point.

It will work properly in the next Cruiser update.

> It seems to be the same logic of changing the planned start into a shaping point if you select "start"

Now rerouting to **Start** sets start as shaping point. Do we want it like this?

**Comment by WolfgangZ42** [comment]
>_It will work properly in the next Cruiser update._

Thanks!

>_Now rerouting to Start sets start as shaping point. Do we want it like this?_

It's not sooo important, but I personally would prefer a via point as original start point, because then I can easily see distance and arrival time to it while navigating ...

**Comment by DolomitesTom** [comment]
Yes via point is better when rerouted to start.

**Comment by devemux86** [comment]
> I personally would prefer a via point as original start point, because then I can easily see distance and arrival time to it while navigating

> via point is better when rerouted to start

I will change it in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
It should work properly in [Cruiser 3.0.9](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
